### ISSUE TITLE
Programs built with 0.4.10 cannot build or run against OpenSSL 1.1.1

### ISSUE THREAD

**Issue by nharward** [softprops/hubcaps#157]
I updated my system recently and now have OpenSSL 1.1.1, discovering shortly afterward that  [ghopac](https://crates.io/crates/ghopac) no longer works. In addition to getting a runtime linker error from the existing binary, I'm also unable to build against the new version. I get a failure compiling openssl-sys v0.9.17 which states the problem:

    This crate is only compatible with OpenSSL 1.0.1, 1.0.2, and 1.1.0, or LibreSSL
    2.5 and 2.6.0, but a different version of OpenSSL was found. The build is now
    aborting due to this version mismatch.

I will try to create a pull request soon, but I thought I'd submit the issue in case someone can get to it before I can.

**Comment by nickelc** [comment]
v0.4.10 builds fine for me with openssl 1.1.1 but i had this issue with reqwest v0.8 and v0.9 fixed the build issues.

**Comment by nharward** [comment]
@nickelc is OpenSSL 1.1.1 the only version you have installed on your machine? I just tried to build master (has an updated Cargo.toml) and it works fine, but switching to the 0.4.10 tag still results in the same error for me. I can build locally from master but I'd like to get a clean build from the published crates.

**Comment by nickelc** [comment]
oh shit sorry, i totally forgot that the hyper 0.12 stuff is not released yet and master is hyper-tls 0.3.

**Comment by nickelc** [comment]
more infos on the issue https://github.com/sfackler/rust-openssl/issues/994

**Comment by softprops** [comment]
Yeah sorry this is such a pain. Is there a newer version of something I could be depending on your help matters?

**Comment by nickelc** [comment]
this [PR](https://github.com/sfackler/rust-openssl/pull/1001) would fix the build issue with openssl 1.1.1 
```diff
diff --git a/Cargo.toml b/Cargo.toml
index a69e814..11186d5 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -37,3 +37,6 @@ version = "0.1.2"
 [features]
 default = ["tls"]
 tls = ["hyper-tls"]
+
+[patch.crates-io]
+openssl = { git = "https://github.com/ishitatsuyuki/rust-openssl", branch = "0.9.x" }
```

**Comment by softprops** [comment]
This should be covered now and with option to opt into rust tls https://github.com/softprops/hubcaps/blob/master/Cargo.toml#L43
